### PR TITLE
update: Be Adaptive for Deepin 20.2

### PR DESCRIPTION
**Pull request by zty199**
fix: Change plugin name to "datetime" to work in Deepin 20.2 properly (All plugins with third-party names have problems);
fix: Fix a bug that font color in speedwidget won't change with system theme color real-time;
feat: Use new tipsWidget in Deepin 20.1, fix some display glitches in MultiLine type.

### CODE REVIEW

File: speedplugin.cpp [hunk 1]
```diff
@@ -42,8 +42,8 @@ SpeedPlugin::SpeedPlugin(QObject *parent)
 
 const QString SpeedPlugin::pluginName() const
 {
-//    return QString("lfxSpeed");  datetime
-    return "lfxSpeed";  // 假装我也叫这个，否则会被压缩，在 1.2.3 版本中才被修改
+//    return QString("lfxSpeed");
+    return "datetime";  // Deepin 20.2 中第三方插件名会导致插件无法正常加载
```
Comment (xmuli) [inline speedplugin.cpp:46]: 此处不建议修改；还是叫 `lfxSpeed ` 比较合适。更名 datetime 的优缺点详细可参见[https://ifmet.cn/posts/c225b552/](https://ifmet.cn/posts/c225b552/) 中 "任务栏 1.2.2 的缺陷" 一节。

此开发环境是 1080 分辨率，然后我调整了默认字体为自带等宽，缩小字体大小，减小间隙，刚好合适，无需更名为 datetime 更加合适。且叫 datetime 会在少见环境下，默认插件不加载。综上，决定更名为 lfxSpeed

File: speedwidget.cpp [hunk 2]
```diff
@@ -69,6 +69,24 @@ SpeedWidget::SpeedWidget(SettingModel *model, QWidget *parent)
     connect(m_model, &SettingModel::sigShowDownChange, this, &SpeedWidget::onShowDown);
     connect(m_model, &SettingModel::sigShowCPUChange, this, &SpeedWidget::onShowCPU);
     connect(m_model, &SettingModel::sigShowMemChange, this, &SpeedWidget::onShowMem);
+
+    connect(DGuiApplicationHelper::instance(), &DGuiApplicationHelper::themeTypeChanged, this, [=]()
```
Comment (xmuli) [inline speedwidget.cpp:73]: 对于白色和黑色主题，我有设置，不过需要重启或者注销之后，才会生效，参见 https://github.com/xmuli/lfxSpeed/blob/4e06b21b97609d880300501abaf862244b36c670/speedwidget.cpp#L109

**Review by XMuli — COMMENTED** [review]
> fix: Change plugin name to "datetime" to work in Deepin 20.2 properly (All plugins with third-party names have problems);
> fix: Fix a bug that font color in speedwidget won't change with system theme color real-time;

It is not necessary to modify the name, see below for details, It is not necessary to change the name, see below for details, but perhaps it can be optimized to customize the theme after switching without restarting (see your connect mode)

> feat: Use new tipsWidget in Deepin 20.1, fix some display glitches in MultiLine type.

I tested it on 20.1 with no problems. Can you attach a screenshot? About showing no line break?

**Comment by xmuli** [comment]
You can try to optimize the following areas

- CPU The actual is slightly off, the correct calculation is as follows: 

https://github.com/xmuli/NetSpeed/blob/886a38808146e368ebe71a36d4d49e602c893b4a/dde-dock-plugin/WinDockNet. cpp#L237


- This version does not consider multi-language, maybe you can help me to change the multi-language (especially where Chinese is used)

- There is no configuration file set to save the changes of this file, after reboot it will be set to m

---

About these points, I have fixed them all in https://github.com/xmuli/NetSpeed, ideas you can also refer to it. If you have questions feel free to leave a comment

**Comment by xmuli** [comment]
But still thank you for submitting. It was a great pleasure. 

image

File: speedplugin.cpp (hunk 1 above)
Comment (zty199) [reply speedplugin.cpp:46]: 20.2 上我做了测试，好像第三方插件名会导致插件只能直接显示在 dock 里面，无法右键选择是否显示，而且插件的右键菜单完全没反应……所以只能先改插件名顶着了……改成 dock 里原有的一些名字就没有这个问题，除了显示顺序会被打乱以外……

**Pull request closed without merging**